File: Selenium/Activities/Activity13.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support.ui import WebDriverWait

driver = webdriver.Chrome()
wait = WebDriverWait(driver, 10)

driver.maximize_window()
driver.get("https://training-support.net/webelements/tables")
print("Title of page is :", driver.title)

rows = driver.find_elements(By.XPATH, "//table/tbody/tr")
print("The number of rows in table is :", len(rows))

columns = driver.find_elements(By.XPATH, "//table/thead/tr/th")
print("The number of columns in table is :", len(columns))

third_rows = driver.find_elements(By.XPATH, "//table/tbody/tr[3]/td")
print("Third row elements")
for element in third_rows:
    print(element.text)

second_row_second_column = driver.find_element(By.XPATH, "//table/tbody/tr[2]/td[2]")
print("The 2nd row and 2 column element:", second_row_second_column.text)

driver.quit()

File: Selenium/Activities/Activity13.java
import java.util.List;
import org.openqa.selenium.By;
import org.openqa.selenium.Keys;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.chrome.ChromeDriver;
import org.openqa.selenium.interactions.Actions;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;
import java.time.Duration;

public class Activity13 {

    public static void main(String[] args) {

        ChromeDriver driver = new ChromeDriver();
        WebDriverWait wait = new WebDriverWait(driver,Duration.ofSeconds(10));
        
        driver.manage().window().maximize();

        driver.get("https://training-support.net/webelements/tables");
        System.out.println("Title of page is : " + driver.getTitle());

        List<WebElement> rows = driver.findElements(By.xpath("//table/tbody/tr"));
        System.out.println("The number of rows in table is :" +rows.size());
        
        List<WebElement> columns = driver.findElements(By.xpath("//table/thead/tr/th"));
        System.out.println("The number of columns in table is :" +columns.size());
      
        List<WebElement> third_rows = driver.findElements(By.xpath("//table/tbody/tr[3]/td"));
        System.out.println("Third row elements");
        for(WebElement element : third_rows) {
        	System.out.println(element.getText());
        	
        }
        	
        
        WebElement second_rows_column = driver.findElement(By.xpath("//table/tbody/tr[2]/td[2]"));
        System.out.println("The 2nd row and 2 column element: "+second_rows_column.getText());
        driver.quit();
    }
}
